### ISSUE TITLE
Memory leak in editor even after calling destroy()

### ISSUE THREAD

**Issue by xaksis** [ueberdosis/tiptap#538]
**Describe the bug**
Memory leak each time an instance of tiptap editor is created (even when the last instance was destroyed)

**Steps to Reproduce / Codesandbox Example**
1. Go to the examples site https://tiptap.scrumpy.io/, you're on Basic example
2. launch the dev console, switch to memory tab and force Garbage collection
3. take a heap snapshot
4. Switch tab to Menu Bubble example then come back to Basic example
5. Force GC then take another snapshot
6. Repeat 4 & 5 a few times
7. You'll see the allocated memory keeps growing with each switch even though editor.destroy is called for each example.

**Expected behavior**
Calling editor.destroy should allow for allocated memory to be freed up and garbage collected

**Screenshots**
<img width="642" alt="Screen Shot 2019-12-06 at 8 32 06 AM" src="https://user-images.githubusercontent.com/4627120/70375363-01739080-18cb-11ea-8963-dbc4d53ed7dc.png">

**Comment by siegerts** [comment]
- Not sure if directly related, but for reference: https://github.com/ProseMirror/prosemirror/issues/660

Also, maybe try combining `editor.destroy()` with [`vm.$off`](https://vuejs.org/v2/api/#vm-off).  May help if somehow related to the above issue/resolution.

**Comment by hanspagel** [comment]
See also #499

**Comment by marcuslindblom** [comment]
Three years later and I'm also experiencing the same issue. @xaksis did you solve this?

**Comment by montella1507** [comment]
I have still problem with this.

I have fixed version of prosemirror to 1.33.2 because there was bug with memory leak regarding Range. 

However, my versions:
```
"prosemirror-view": "1.33.2",
    "@tiptap/core": "^2.0.0-beta.217",
    "@tiptap/extension-bubble-menu": "^2.0.0-beta.217",
    "@tiptap/extension-color": "^2.0.0-beta.217",
    "@tiptap/extension-floating-menu": "^2.0.0-beta.217",
    "@tiptap/extension-font-family": "^2.0.0-beta.217",
    "@tiptap/extension-heading": "^2.0.0-beta.217",
    "@tiptap/extension-highlight": "^2.1.12",
    "@tiptap/extension-link": "^2.0.0-beta.217",
    "@tiptap/extension-text": "^2.0.0-beta.217",
    "@tiptap/extension-text-align": "^2.0.0-beta.217",
    "@tiptap/extension-text-style": "^2.0.0-beta.217",
    "@tiptap/extension-typography": "^2.0.0-beta.217",
    "@tiptap/extension-underline": "^2.0.0-beta.217",
    "@tiptap/pm": "^2.0.0-beta.217",
    "@tiptap/starter-kit": "^2.0.0-beta.217",
```


but the Editor keeps my elements in memory:
<img width="1768" alt="image" src="https://github.com/ueberdosis/tiptap/assets/10418323/23b8c973-c0b7-4057-bd02-2b23e0d0e18b">
